### PR TITLE
Debug navbar code to fix logo broken link

### LINKED ISSUE #323: Icon on upper left corner of Astropy tutorial pages appears with broken link to source
When I was browsing the Astropy Learn tutorials as on February 3rd, 2019, I see the following on my computer screen:

screenshot 2019-02-03 00 35 54

The Astropy icon seems to be not showing due to broken link to source image file... Is anyone else experiencing the same issue? If so, should I open a PR to correct this? This issue could be related to Issues #310. Please advice.

### PR DESCRIPTION
**Pull request by kakirastern**
Fixes #323. 

Attempts to mend broken Astropy logo banner link on the tutorial pages, in order for the image on the left hand side of the Learn Astropy navbar to show properly.

### CODE REVIEW

**Comment by agarwalrounak** [comment]
I don't think this fixes the issue. What you have done is change the `src` of `astropy_logo_banner.png` whereas the image on the left hand side of the Learn Astropy navbar is `astropy_logo.png`

You probably have to make changes in line 44 of `navbar.html`

**Comment by kakirastern** [comment]
Thanks @agarwalrounak! Was a bit too rash and brash, and have neglected to check everything thoroughly. Thanks for pointing it out. A second commit's been made to render everything internally consistent.

**Comment by kakirastern** [comment]
Hi @agarwalrounak, would you like to check if further changes needed to be made in this PR, and let me know if there is any suggestion to improve on the commits?

**Comment by agarwalrounak** [comment]
I don't think changing the image source will solve the issue. Have you tried to test your changes locally?

I have my exams going on and haven't got the time to set up the project on my system. Otherwise, I would have been able to help you in this regard.

**Comment by kakirastern** [comment]
Hi @agarwalrounak! Yup, I am in the process of testing the changes locally. But will take a while, as this is my first time. I was counting on giles to do the work for me, but now that it's gone I cannot rely on it anymore. 

No worries, just wanted to see if you would like to contribute more in order to prepare for GSoC 2019 if OpenAstronomy will be selected again as a mentoring organization. All the best on your exams!

**Comment by kakirastern** [comment]
Cannot fix the broken logo link yet, but did debugged other issues with the code. Now the local build works perfect.

**Comment by kakirastern** [comment]
@MananAgarwal Turned out the sidebars including the `navbar` got directed to the "build/html/rst-tutorials/" directory automatically instead of the desired "build/html/" directory, so the logo path link was broken. I "tricked" the system with the path "../_static/astropy_logo.png" to restore the link to the source of the image. Anyway it's my poor man's way to fixing things for now. Also, I found other bugs in the themes/template/ code files, and have corrected them in this PR. You might want to look into it further, or not. 

The logo on the upper left corner is finally showing with external linkage to [http://www.astropy.org/astropy-tutorials/](http://www.astropy.org/astropy-tutorials/) now.

**Comment by MananAgarwal** [comment]
Hey @kakirastern, It's good to see you work so hard to solve this issue.
Even I had thought of this approach to "trick" the system, but this comes with a loss of generality. You are assuming that the "_static" folder is always in the parent directory of the current folder, which is true for the tutorials (at least in the current file structure). Your approach will break when you open pages like say, "contributing.html" or "dev.html" etc, as these pages are in the same directory as "_static" folder.

Other changes that you have made in the template code files are semantic changes. They definitely improve the consistency of the codebase. Thanks for pointing these out.

**Comment by kakirastern** [comment]
Hey @MananAgarwal, thanks for the reply! Sure, at least things work now as far as the upper left corner logo is concerned... Hopefully this quick fix will hold up for a while.

**Comment by MananAgarwal** [comment]
@kakirastern maybe I wasn't clear. This implementation is breaking in some pages.
Check "contributing.html" ([link](https://342-8307899-gh.circle-artifacts.com/0/home/circleci/project/build/html/contributing.html) to latest giles build)

**Comment by kakirastern** [comment]
Yup, have worked out a functional fix...

**Comment by kakirastern** [comment]
I have a feeling we are on the right track, as the real bugs in the tutorials are finally showing... am working on these so as to pass Circle CI and Travis CI one-by-one.

**Comment by kakirastern** [comment]
Everything is okay for now. Will see if this can be merged soon. Thanks @agarwalrounak and @MananAgarwal!

File: tutorials/conf.py [hunk 1]
```diff
@@ -96,7 +96,12 @@
 # The name of an image file (within the static path) to use as favicon of the
 # docs.  This file should be a Windows icon file (.ico) being 16x16 or 32x32
 # pixels large.
-html_favicon = 'astropy_favicon.ico'
+html_favicon = "_static/astropy_favicon.ico"
```
Comment (adrn) [inline tutorials/conf.py]: This shouldn't be changed: if you look at the comment above this block, it says "the name of an image file (within the static path)", so in this case you don't need to preface it with `_static`

File: tutorials/conf.py [hunk 2]
```diff
@@ -96,7 +96,12 @@
...
+
+# (Optional) Logo. Should be small enough to fit the navbar (ideally 24x24).
```
Comment (adrn) [inline tutorials/conf.py]: Can you explain this addition? With our current theme, I don't think we need to set this, so I would suggest removing this.

File: tutorials/notebooks/Coordinates-Intro/Coordinates-Intro.ipynb [hunk 3]
```diff
@@ -45,6 +45,7 @@
     "from astropy import units as u\n",
     "from astropy.coordinates import SkyCoord\n",
     "from astropy.table import Table\n",
```
Comment (adrn) [inline tutorials/notebooks/Coordinates-Intro/Coordinates-Intro.ipynb]: The changes in this file seem unrelated to this PR.

File: tutorials/notebooks/coordinates/coordinates.ipynb [hunk 4]
```diff
@@ -30,6 +30,7 @@
     "from astropy import units as u\n",
     "from astropy.coordinates import SkyCoord\n",
     "from astropy.table import Table\n",
+    "from astropy.utils import data\n",
```
Comment (adrn) [inline tutorials/notebooks/coordinates/coordinates.ipynb]: The changes in this file also seem unrelated to this PR.

File: tutorials/themes/tutorials-theme/layout.html [hunk 5]
```diff
@@ -27,7 +27,7 @@
 <!DOCTYPE html>
 {%- endblock %}
```
Comment (adrn) [inline tutorials/themes/tutorials-theme/layout.html]: Is this a stylistic change? And are we sure both names are valid?

File: tutorials/themes/tutorials-theme/navbar.html [hunk 6]
```diff
@@ -39,9 +39,10 @@ <h2 class="card-title text-center"><a href="{{ pathto('examples') }}" class="tex
 </div>
 
 {% else %}
-
 <nav class="navbar navbar-expand-lg bg-navbar">
-  <a class="navbar-brand" href="/astropy-tutorials"style="padding: 0px 5px;"><img src="{{ pathto('_static/' + 'astropy_logo.png') }}" style="height:38px;"></a>
```
Comment (adrn) [inline tutorials/themes/tutorials-theme/navbar.html]: Instead of hard-linking to `astropy.org/astropy-tutorials`, I would instead recommend changing this to:

```html
<a class="navbar-brand" href="/astropy-tutorials"style="padding: 0px 5px;">
<img src="{{ pathto('_static/' + 'astropy_logo.png', 1) }}" style="height:38px;">
</a>
```

[I just learned here](https://www.sphinx-doc.org/en/master/templating.html) that `pathto(file)` links to an HTML document (so it was breaking the link), but `pathto(file, 1)` will link to a static file (i.e. an image, as we want here), so this is what we want.

File: tutorials/themes/tutorials-theme/theme.conf [hunk 7]
```diff
@@ -44,7 +44,7 @@ globaltoc_includehidden = true
 
 # HTML navbar class (Default: "navbar") to attach to <div> element.
 # For black navbar, do "navbar navbar-inverse"
-navbar_class = navbar
+navbar_class = navbar navbar-inverse
```
Comment (adrn) [inline tutorials/themes/tutorials-theme/theme.conf]: Can you explain this addition?

File: tutorials/conf.py (hunk 1 above)
Comment (kakirastern) [reply tutorials/conf.py]: I tried it without the `_static/` path and an error was thrown. And I double-checked on the Internet and someone mentioned that it should be relative to the confdir location, and not relative to the `_static/` path... Could you double check for me, maybe it has to do with the `html_path_static` path defined after this line? I built the docs locally and if I do what you are suggested this file cannot be found.

File: tutorials/conf.py (hunk 2 above)
Comment (kakirastern) [reply tutorials/conf.py]: Yup, sure... we don't need it. I was playing around with it and omitted to delete it.

File: tutorials/notebooks/Coordinates-Intro/Coordinates-Intro.ipynb (hunk 3 above)
Comment (kakirastern) [reply tutorials/notebooks/Coordinates-Intro/Coordinates-Intro.ipynb]: I was trying to get Circle CI and Travis CI passed, since some errors related to this file was thrown after I fixed the broken logo link.

File: tutorials/notebooks/coordinates/coordinates.ipynb (hunk 4 above)
Comment (kakirastern) [reply tutorials/notebooks/coordinates/coordinates.ipynb]: Similar to previous comment, I was trying to get Circle CI and Travis CI passed, since some errors related to this file was thrown after I fixed the broken logo link.

File: tutorials/themes/tutorials-theme/theme.conf (hunk 7 above)
Comment (kakirastern) [reply tutorials/themes/tutorials-theme/theme.conf]: We have a dark navbar, so navbar-inverse supposedly should be added. Just in case.

File: tutorials/themes/tutorials-theme/navbar.html (hunk 6 above)
Comment (kakirastern) [reply tutorials/themes/tutorials-theme/navbar.html]: Sure, that would be better...

File: tutorials/themes/tutorials-theme/layout.html (hunk 5 above)
Comment (kakirastern) [reply tutorials/themes/tutorials-theme/layout.html]: I changed `navBar` to `nav_bar` so that there will definitely be no confusion in case the system used is not case sensitive.

**Comment by kakirastern** [comment]
I can rebase and squash if necessary... Please advise if any other changes need to be made.

File: tutorials/conf.py (hunk 1 above)
Comment (adrn) [reply tutorials/conf.py]: Strange - this does remove the WARNING but I don't understand why it seems to work even without it. In any case, keep it!

File: tutorials/notebooks/Coordinates-Intro/Coordinates-Intro.ipynb (hunk 3 above)
Comment (adrn) [reply tutorials/notebooks/Coordinates-Intro/Coordinates-Intro.ipynb]: No problem, but let's fix the travis build in a separate PR, since this is a change to the tutorial *content*

File: tutorials/themes/tutorials-theme/navbar.html [hunk 8]
```diff
@@ -39,9 +39,10 @@ <h2 class="card-title text-center"><a href="{{ pathto('examples') }}" class="tex
...
+  <a class="navbar-brand" href="http://www.astropy.org/astropy-tutorials/" style="padding: 0px 5px;">
```
Comment (adrn) [inline tutorials/themes/tutorials-theme/navbar.html]: We don't want to assume the domain here, so change to `href="/astropy-tutorials"`

File: tutorials/conf.py (hunk 1 above)
Comment (kakirastern) [reply tutorials/conf.py]: Sure, no problem!

File: tutorials/conf.py (hunk 2 above)
Comment (kakirastern) [reply tutorials/conf.py]: Removed.

**Comment by kakirastern** [comment]
@adrn Commits rebased and squashed into one. Okay to merge now. Please double check if necessary.

**Comment by kakirastern** [comment]
@adrn I will open another PR to address the "bugs" showing up on Travis...

File: tutorials/conf.py [hunk 9]
```diff
@@ -96,7 +96,11 @@
...
```
Comment (adrn) [inline tutorials/conf.py]: One more thing: please remove these commented-out lines, as we decided they are unnecessary. Thanks!

Comment (kakirastern) [reply tutorials/conf.py]: Yup, removed...

**Comment by adrn** [comment]
Thanks @kakirastern!

**Comment by kakirastern** [comment]
You are welcome, @adrn!

**Comment by kakirastern** [comment]
Solution not working completely. Have opened PR #339 to address that.

**Comment by kakirastern** [comment]
Sorry, my bad, it works now... Was definitely too rash and brash.
